File: August-LeetCoding-Challenge/06-Find-All-Duplicates-in-an-Array/Find-All-Duplicates-in-an-Array.cs
public class Solution {
    public IList<int> FindDuplicates(int[] nums) {
        var res = new List<int>();
        for(int i = 0; i<nums.Length; i++){
            var idx = Math.Abs(nums[i]) -1;
            if(nums[idx] < 0){
                res.Add(idx+1);
            }
            nums[idx] = -nums[idx];
        }
        return res;
    }
}

File: August-LeetCoding-Challenge/06-Find-All-Duplicates-in-an-Array/Find-All-Duplicates-in-an-Array.kt
package string_integer

import java.util.*
import kotlin.math.abs

class FindAllDuplicatesinanArrayKotlin442 {
    fun findDuplicates(nums: IntArray): List<Int> {
        val results: MutableList<Int> = LinkedList()
        nums.forEach {
            val index = abs(it) - 1
            if (nums[index] < 0) {
                results.add(index + 1)
            }
            nums[index] = -nums[index]
        }
        return results
    }
    /*
    fun findDuplicates(nums: IntArray): List<Int> {
        val results: MutableList<Int> = LinkedList()
        val set: MutableSet<Int> = HashSet()
        nums.forEach {
            if (set.contains(it)) {
                results.add(it)
            }
            set.add(it)
        }
        return results
    }
     */
}